packages/@aws-cdk/aws-lambda/lib/layers.ts

```diff
@@ -0,0 +1,237 @@
+import cdk = require('@aws-cdk/cdk');
+import { Code } from './code';
+import { Runtime } from './runtime';
+
+export interface LayerVersionProps {
+  /**
+   * The runtimes that this layer is compatible with.
+   *
+   * @default All runtimes are supported
+   */
+  compatibleRuntimes?: Runtime[];
+
+  /**
+   * The content of this Layer. Using the *inline* code is not permitted.
+   */
+  code: Code;
+
+  /**
+   * The description the this Lambda Layer.
+   */
+  description?: string;
+
+  /**
+   * The SPDX licence identifier or URL to the license file for this layer.
+   */
+  license?: string;
+
+  /**
+   * The name of the layer.
+   * @default a name will be generated.
+   */
+  name?: string;
+}
+
+export interface ILayerVersion {
+  /**
+   * The ARN of the Lambda Layer version that this Layer defines.
+   */
+  readonly layerVersionArn: string;
+
+  /**
+   * The runtimes compatible with this Layer.
+   */
+  readonly compatibleRuntimes?: Runtime[];
+
+  /**
+   * Grants usage of this layer to specific entities. Usage within the same account where the layer is defined is always
+   * allowed and does not require calling this method. Note that the principal that creates the Lambda function using
+   * the layer (for example, a CloudFormation changeset execution role) also needs to have the
+   * ``lambda:GetLayerVersion`` permission on the layer version.
+   *
+   * @param grantee the identification of the grantee.
+   * @param grantId the ID of the grant in the construct tree.
+   */
+  grantUsage(grantee: LayerVersionUsageGrantee, grantId: string): ILayerVersion
```

Comment:
Switch parameter order here. Id first, props second. Looks more like construct instantiations and other sites where we also need an ID.